modules/renter/contractor/journal.go

```diff
@@ -0,0 +1,492 @@
+package contractor
+
+// The contractor achieves efficient persistence using a JSON transaction
+// journal. It enables efficient ACID transactions on JSON objects.
+//
+// Each journal represents a single JSON object. The object is serialized as
+// an "initial object" followed by a series of update sets, one per line. Each
+// update specifies a field and a modification. See the journalUpdate type for
+// a full specification.
```

Comment:
In a full WAL, updates serialized into json in a batchable way sounds quite reasonable.

I think we'll be able to keep a lot of this code.